File: project/cs/v3cs/v1cs/Librariar.cs
/***********************************************************************
 * Module:  Librariar.cs
 * Author:  Nabra
 * Purpose: Definition of the Class Librariar
 ***********************************************************************/

using System;

enum worcker { readroom, abonement }; //місце роботи

public class Librariar
{
   public void LoanBook()
   {
      // TODO: implement
   }
   
   public void PotBookBeck()
   {
      // TODO: implement
   }
   
   public void GiveServises()
   {
      // TODO: implement
   }
   
   public void GiveWiFi()
   {
      // TODO: implement
   }
    /// <summary>
    /// Конструктор за замовченням
    /// </summary>
    public	Librariar()//конструктор за замовченням
   {
       Groop = worcker.abonement;//місце роботи - абонемент
       Console.WriteLine("Librariar created standartly");
   }
    /// <summary>
    /// Конструктор перетворення типів
    /// </summary>
    /// <param name="i">Код місця роботи бібліотекаря</param>
	public Librariar(int i)//конструктор перетворення типів
    {
        Groop = (worcker)(i % 2);//код місця роботи - попередньо введене число
        Console.WriteLine("Librariar created by type change");
    }
    /// <summary>
    /// Конструктор копії
    /// </summary>
    /// <param name="libr">Об*єкт, що копіюється</param>
    public Librariar(Librariar libr)//конструктор копії
    {
        Groop = libr.Groop;
        Console.WriteLine("Librariar created by copy");
    }
    private worcker Groop { get; set; }

}

File: project/c++/V2/Version1/Version1/Librariar.cpp
/***********************************************************************
 * Module:  Librariar.cpp
 * Author:  Nabra
 * Modified: 20 октября 2016 г. 1:45:44
 * Purpose: Implementation of the class Librariar
 ***********************************************************************/

#include "Librariar.h"
#include <iostream>

using namespace std;

//=========== Конструктор за замовченням =========== 
Librariar::Librariar()
{
	groop = abonement;//місце роботи - абонемент
	cout << "Librariar created standartly" << endl;
}

//=========== Конструктор перетворення типів =========== 
Librariar::Librariar(int i)
{
	groop = (worcker)(i%2);//код місця роботи - попередньо введене число
	cout << "Librariar created by type change" << endl;
}

//=========== Конструктор копії =========== 
Librariar::Librariar(const Librariar &libr)
{
	groop = libr.groop;
}

//=========== Деструктор =========== 
Librariar::~Librariar()
{
	cout << "Librariar destroed" << endl;
}

////////////////////////////////////////////////////////////////////////
// Name:       Librariar::LoanBook()
// Purpose:    Implementation of Librariar::LoanBook()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Librariar::LoanBook(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Librariar::PotBookBeck()
// Purpose:    Implementation of Librariar::PotBookBeck()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Librariar::PotBookBeck(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Librariar::GiveServises()
// Purpose:    Implementation of Librariar::GiveServises()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Librariar::GiveServises(void)
{
   // TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Librariar::GiveWiFi()
// Purpose:    Implementation of Librariar::GiveWiFi()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Librariar::GiveWiFi(void)
{
   // TODO : implement
}